File: js-pr/56/javascript/code-quality-readability-maintainability/number_processor_33.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
function processNumberList(numbers) {
    return numbers.map(number => {
        if (number % 2 === 0) {
            return processEvenNumber(number);
        } else {
            return processOddNumber(number);
        }
    });
}

function processEvenNumber(number) {
    if (number > 10 && number < 100) {
        return number * 2;
    } else if (number >= 100) {
        return number / 2;
    } else {
        return number + 1;
    }
}

function processOddNumber(number) {
    return number > 5 ? number - 1 : number + 3;
}
// {/fact}

File: python-pr/52/python/code-quality-readability-maintainability/number_processor_33.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
def process_number_list(numbers):
    processed_numbers = []
    
    for number in numbers:
        if number % 2 == 0:  # Even numbers
            processed_numbers.append(
                number * 2 if 10 < number < 100 else number / 2 if number >= 100 else number + 1
            )
        else:  # Odd numbers
            processed_numbers.append(
                number - 1 if number > 5 else number + 3
            )
    
    return processed_numbers
# {/fact}